### PR TITLE
Refactor NNI manager globals (step 1) - argparse

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Description ###

Refactor NNI manager's argparse with de facto standard library `yargs`.

Changed args and options:

  + `start_mode` -> `action` because "mode" has too many meanings in NNI manager.
      - `new` -> `create` because the command is called "nnictl create".
      - `resume + readonly` -> `view` because the command is called "nnictl view".
  + `log_dir` -> `experiments-directorey` because it is not log dir at all.
      - In config file it is called `experimentWorkingDirectory` but... it was a mistake. I thought it was `~/nni-experiments/<ID>` but no, it is just `~/nni-experiments`.

### Checklist ###
  - [x] test case
  - [x] doc

### How to test ###

Test together with following up PRs.

### CODE REVIEW

**Review by littlelittlehorse — APPROVED** [review]

**Comment by J-shang** [comment]
> ### Description
> Refactor NNI manager's argparse with de facto standard library `yargs`.
> 
> Changed args and options:
> 
> * `start_mode` -> `action` because "mode" has too many meanings in NNI manager.
>   
>   * `new` -> `create` because the command is called "nnictl experiment create".
>   * `resume + readonly` -> `view` because the command is called "nnictl view". (Why it's not "nnictl experiment view"? 🤔)
> * `log_dir` -> `experiments-directorey` because it is not log dir at all.
>   
>   * In config file it is called `experimentWorkingDirectory` but... it was a mistake. I thought it was `~/nni-experiments/<ID>` but no, it is just `~/nni-experiments`.
> 
> ### Checklist
> * [x]  test case
> * [x]  doc
> 
> ### How to test
> Test together with following up PRs.

the command is `nnictl create` and we don't have `nnictl experiment create` 😂
maybe `experimentsWorkingDirectory`?

**Comment by liuzhe-lz** [comment]
> the command is `nnictl create` and we don't have `nnictl experiment create` 😂 maybe `experimentsWorkingDirectory`?

Ooops, seems there was a bug in my mind.
`experimentsWorkingDirectory` sounds better. I will update.

File: nni/experiment/launcher.py [hunk 1]
```diff
@@ -55,20 +55,12 @@ def __init__(self, action, exp_id, config, port, debug, foreground, url_prefix):
...
-        else:
-            self.start_mode = 'new'
-
     def to_command_line_args(self):
         ret = []
         for field in fields(self):
             value = getattr(self, field.name)
             if value is not None:
-                ret.append('--' + field.name)
+                ret.append('--' + field.name.replace('_', '-'))
```
Comment (QuanluZhang) [inline nni/experiment/launcher.py:63]: why this update?

**Review by J-shang — APPROVED** [review]

**Review by QuanluZhang — APPROVED** [review]

**Comment by J-shang** [comment]
There is a conflict, pls fix it

**Pull request merged**